source/server/hot_restart_nop_impl.h

```diff
@@ -14,7 +17,18 @@ namespace Server {
  */
 class HotRestartNopImpl : public Server::HotRestart {
 public:
-  HotRestartNopImpl() {}
+  HotRestartNopImpl() {
+    int num_stats = 100;
+    const MemoryHashSetOptions options = memHashOptions(num_stats);
+    ptr_ = static_cast<uint8_t*>(
+        calloc(MemoryHashSet<Stats::RawStatData>::numBytes(options), num_stats));
+    stats_allocator_ =
+        std::make_unique<Stats::BlockRawStatDataAllocator>(options, true, ptr_, &stat_lock_);
```

Comment:
@mrice32 when you see new statLock / stat_lock_ objects in HotRestart, HotRestartImpl, HotRestartNopImpl, and HotRestart mock objects, it's because of this line. I wanted to create and use a lock mechanism even when the restart isn't hot.

Reply:
But why does it need to have its own accessor method?